packages/ntcore-ts-client/src/lib/pubsub/pubsub.ts

```diff
@@ -53,6 +57,40 @@ export class PubSubClient {
     return instance;
   }
 
+  /**
+   * Adds a listener for when topics are announced.
+   * @param callback - A callback triggered with the announced topic
+   * @param announceAll - Whether to trigger all topics in NetworkTables to be announced
+   * @returns A function to remove the listener.
+   */
+  addOnTopicAnnouncedListener(callback: (_: NetworkTablesTopic) => void, announceAll?: boolean) {
+    this.onTopicAnnouncedListeners.push(callback);
+    if (announceAll) {
+      const subscribeParams: SubscribeMessageParams = {
+        topics: [''],
+        subuid: Util.generateUid(),
+        options: {
+          prefix: true,
+          topicsonly: true,
+        },
+      };
+      this.messenger.subscribe(subscribeParams);
+    }
+    return () => {
+      this.onTopicAnnouncedListeners = this.onTopicAnnouncedListeners.filter(
+        (storedCallback) => storedCallback !== callback
+      );
+    };
+  }
+
+  /**
+   * Notifies all onTopicAnnounced subscribers of the announced topic.
+   * @param topic - The topic that was announced
+   */
+  private notifyOnTopicAnnouncedSubscribers(topic: NetworkTablesTopic) {
```

Comment:
I like this, but I think we could do some extra "syntactic sugar" so to speak to do the extra step of actually subscribing to every new topic if the user wants to.

It would be a new listener that:
1. finds every current topic
2. subscribes to every current topic
3. sets up a listener for new topics and auto subscribes

There's probably some considerations that need to be taken since this is asynchronous... what happens if a new topic is announced as we are subscribing to current topics?

> note to self: I need to refresh myself with the protocol to see if there's bulk subscription messages we could send to the server